main.go

```diff
@@ -125,3 +199,62 @@ func actionSetLogFile(_ *cli.Context, path string) error {
 
 	return nil
 }
+
+func profilePath(nameFormat string) string {
+	filename := fmt.Sprintf(nameFormat, os.Getpid())
+	if profilePrefix != "" {
+		filename = filepath.Join(profilePrefix, filename)
```

Comment:
It bothers me that this function implies there is some flag ordering incidence -- the actions are called as soon as the flags is seen/processed, so if the `-profile-prefix` flags is provided after a profiler selection flag, it won't be honored.

Reply:
Indeed. Moved this all to `Before` as suggested in your other comment, which as I understand it will run after all of the args are processed.